pkg/corerp/frontend/controller/applications/graph_util.go

```diff
@@ -547,61 +551,61 @@ func findSourceResource(source string, allResources []generated.GenericResource)
 	return orig, ErrInvalidSource
 }
 
-// providesFromAPIData is specifically to support HTTPRoute.
-func providesFromAPIData(resource generated.GenericResource) []*corerpv20231001preview.ApplicationGraphConnection {
-	// Any Radius resource type that exposes a port uses the following property path to return them.
-	// The port may have a 'provides' attribute that specifies a httproute.
-	// This route should be parsed to find the connections between containers.
-	// For example, if container A provides a route and container B consumes it,
-	// then we have port.provides in container A and container.connection in container B.
-	// This gives us the connection: container A --> route R --> container B.
-	// Without parsing the 'provides' attribute, we would miss the connection between container A and route R.
-
-	p, err := jsonpointer.New(portsPath)
-	if err != nil {
-		// This should never fail since we're hard-coding the path.
-		panic("parsing JSON pointer should not fail: " + err.Error())
+// connectionsConverter resolves the outbound connections for a resource from the generic resource representation.
+// For example if the resource is an 'Applications.Core/container' then this function can find it's connections
+// to other resources like databases. Conversely if the resource is a database then this function
+// will not find any connections (because they are inbound). Inbound connections are processed later.
+func connectionsConverter(resources []generated.GenericResource) func(any) (string, corerpv20231001preview.Direction, error) {
+	return func(item any) (string, corerpv20231001preview.Direction, error) {
+		data := &corerpv20231001preview.ConnectionProperties{}
+		err := toStronglyTypedData(item, data)
+		if err != nil {
+			return "", "", err
+		}
+		sourceID, err := findSourceResource(to.String(data.Source), resources)
+		if err != nil {
+			return "", "", err
+		}
+		return sourceID, corerpv20231001preview.DirectionOutbound, nil
 	}
+}
 
-	raw, _, err := p.Get(&resource)
-	if err != nil {
-		// Not found, this is fine.
-		return []*corerpv20231001preview.ApplicationGraphConnection{}
+func routesPathConverter(resources []generated.GenericResource) func(any) (string, corerpv20231001preview.Direction, error) {
+	return func(item any) (string, corerpv20231001preview.Direction, error) {
```

Comment:
Can't we have one function for all three and just pass in the resource type?

Reply:
> Can't we have one function for all three and just pass in the resource type?

That way makes more complicated and each one has different logic for scanning resource. it is not worth making.